add_heading_anchors.js

```diff
@@ -0,0 +1,40 @@
+var addHeadingAnchors = {
+
+  init: function (selector) {
+    this.target = document.querySelector(selector);
+    if (this.target) {
+      this.addAnchorsToHeadings();
+    }
+  },
+
+  addAnchorsToHeadings: function () {
+    var headings = this.target.querySelectorAll('h1, h2, h3, h4, h5, h6'),
+      id;
+
+    headings.forEach(function (heading) {
+      if (id = heading.id) {
```

Comment:
This is supposed to be shorthand for 

```
if (heading.id) {
id = headingid;
....
}

```

right?  At first glance it's not obvious that it's not a typoed comparison.  This also may be redundant if the suggested selector works.